### ISSUE TITLE
Add `Operation` class to generated umbrella app

### ISSUE THREAD

**Issue by timriley** [dry-rb/dry-web-roda#43]
- include Dry::Monads::Either::Mixin
- include Dry::Matcher.for(:call, with: Dry::Matcher::EitherMatcher)

**Comment by solnic** [comment]
Hmm I'm wondering whether it'd make sense to add `Operation` class to `dry-transaction`. Folks could just do `class DoSomething < Dry::Transaction::Operation`. WDYT?

**Comment by GustavoCaso** [comment]
Sorry to interrupt but what do you mean with `Operation` are talking about Transaction operations like `step`, `tee`, `map` etc...

**Comment by timriley** [comment]
@GustavoCaso we're talking about typical "functional objects" here, the kind you build hundreds of when you're making a large app (hence the idea of having the basic functionality provided by a shared superclass). For example

```ruby
```

**Comment by timriley** [comment]
@GustavoCaso we're talking about typical "functional objects" here, the kind you build hundreds of when you're making a large app (hence the idea of having the basic functionality provided by a shared superclass). For example

```ruby
class CreateComment < Some::Shared:Operation
  def call(attrs)
    # ...
  end
end
```

@solnic nice idea to put it in dry-transaction. The reason I didn't initially consider providing this via a gem was that the notion of what an "operation" is might differ from app to app, so by dry-web-roda generating one for you, it was more of a hint that you should consider it for your own app.

That said, if all we'd be looking to do initially is include those 2 mixins, then it does make sense that this could be a `Dry::Transaction::Operation`. And that might make adoption of operations/transactions just that little bit easier for people, too.

OK! So if we did that, I still have one question: even if `Dry::Transaction::Operation` exists, is there still merit in generating something like this in new dry-web-roda apps?

```ruby
module MyApp
  class Operation < Dry::Transaction::Operation
    # shared logic goes here
  end
end
```

I guess if the class if going to be empty, maybe not, but perhaps worth having as a placeholder just in case there's some shared behaviour that an app wants to provide?

**Comment by GustavoCaso** [comment]
Thanks @timriley for explaining 😄 

So if understood correctly `Dry::Transaction::Operation` will be such an interface to `MyApp::Operation` adding some shared method across all operations?

**Comment by timriley** [comment]
@GustavoCaso At this point we're basically just talking about this:

```ruby
module Dry
  module Transaction
    class Operation
      include Dry::Monads::Either::Mixin
      include Dry::Matcher.for(:call, with: Dry::Matcher::EitherMatcher)
    end
  end
end
```

😄

**Comment by GustavoCaso** [comment]
Nice I got it 👍

**Comment by solnic** [comment]
> OK! So if we did that, I still have one question: even if Dry::Transaction::Operation exists, is there still merit in generating something like this in new dry-web-roda apps?

Yes, I would say it should be a general good practice (just like we do with rom repos), because it gives you a place to inject stuff like loggers, notifiers, integrate with anything that's specific to your app. It also makes it easier to swap implementations for many classes once there's just one abstract class that you own.

**Comment by timriley** [comment]
Fixed in #45
